src/v3/src/hooks/useInteractionCodeFlow.ts

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright (c) 2023-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import { AuthSdkError, IdxStatus, IdxTransaction } from '@okta/okta-auth-js';
+import { useEffect, useState } from 'preact/hooks';
+
+import { RenderResult } from '../../../types';
+import { getTypedOAuthError, RecoverableError } from '../../../util/OAuthErrors';
+import { redirectTransformer } from '../transformer/redirect';
+import { createForm } from '../transformer/utils';
+import { FormBag, InfoboxElement, WidgetProps } from '../types';
+import {
+  getMode, isAuthClientSet, isOauth2Enabled, loc,
+} from '../util';
+
+export const useInteractionCodeFlow = (
+  transaction: IdxTransaction | undefined,
+  widgetProps: WidgetProps,
+  onSuccess?: WidgetProps['globalSuccessFn'],
+  onError?: WidgetProps['globalErrorFn'],
+): FormBag | undefined => {
+  const [formBag, setFormBag] = useState<FormBag | undefined>();
+
+  if (!isAuthClientSet(widgetProps)) {
+    throw new Error('authClient is required');
+  }
+
+  useEffect(() => {
+    if (typeof transaction === 'undefined') {
+      setFormBag(undefined);
+      return;
+    }
+
+    if (!([IdxStatus.TERMINAL, IdxStatus.SUCCESS].includes(transaction.status)
+        || !transaction.nextStep)) {
+      setFormBag(undefined);
+      return;
+    }
+
+    if (!(isOauth2Enabled(widgetProps) && transaction.interactionCode)) {
+      setFormBag(undefined);
+      return;
+    }
+
+    const {
+      authClient,
+      redirectUri,
+      redirect,
+    } = widgetProps;
+    const { interactionCode } = transaction;
+    const transactionMeta = authClient.idx.getSavedTransactionMeta();
+    const state = authClient.options.state || transactionMeta?.state;
+    const redirectParams = { interaction_code: interactionCode || '', state: state || '' };
+
+    const isRemediationMode = getMode(widgetProps) === 'remediation';
+    if (isRemediationMode) {
+      authClient.idx.clearTransactionMeta();
+    }
+
+    const shouldRedirect = redirect === 'always';
+    if (shouldRedirect) {
+      if (!redirectUri) {
+        throw new Error('redirectUri is required');
+      }
+
+      const searchParams = new URLSearchParams(redirectParams);
+      const redirectFormBag = redirectTransformer(
+        transaction,
+        `${redirectUri}?${searchParams.toString()}`,
```

Comment:
if `interaction_code` or `state` in redirectParams are empty strings, would it still show up in the search params of the redirect uri with empty values?

Reply:
Actually, in this case interaction_code will never be undefined so I am updating that to remove the `|| ''`. Regarding state, based on the types it _could_ be undefined but in the real world, it would never be undefined. However, I will update the logic to properly account for this. 

Thank you!